### DISCUSSION TITLE
Risques légaux de publier la clef d'encryptage

### DISCUSSION THREAD

**Discussion by fparrel** [marian-m12l/studio#368] [Q&A]
Bonjour
Suite à la 5ème casse de l'appareil de mes enfants, je n'arrive plus à ressouder la molette, et j'ai du coup développé un lecteur sous Android: j'insère la SDCard dans la tablette, et j'ai une interface pour lire les histoires.
Je me suis basé sur le code de studio (sans ca, je ne me serais pas lancé là dedans) et j'ai réutilisé du coup la même license MPL2.0
Je m'interroge sur les risques légaux encourus. Dans le code source de studio, et du coup dans le mien, il y a la clef d'encryptage commune à tous les appareils. Quelqu'un a-t-il des renseignements sur les risques encourus? Je suppose que la clef a été obtenue via téléchargement du binaire via JTAG et décompilation, et c'est une clef symétrique, donc on peut dire qu'elle était pas vraiment bien cachée puisque accessible à tout développeur compétent et motivé.

**Comment by marian-m12l** [comment]
Bonjour,

Content de voir que STUdio est utile pour de nouveaux projets :slightly_smiling_face: .
Pour la licence, je n'ai plus tout en tête mais il me semble que la MPL n'impose pas de réutiliser la même licence.

Merci pour la question, qui va me permettre de clarifier le sujet.


Effectivement, la clef a été obtenue par rétro-ingénierie. Tout comme l'algorithme de chiffrement, le format des histoires et les commandes USB pour la v1 (qui n'apparaît pas comme un stockage de masse).


Je ne suis pas juriste, mais voici le raisonnement qui justifie la légalité de ce projet :

1) STUdio est un logiciel créé de façon indépendante ;
2) STUdio n'est pas un logiciel concurrent au Luniistore, mais apporte un nouvel usage : la création d'histoires interactives par l'utilisateur ;
3) Les éléments nécessaires à l'interopérabilité avec l'appareil et le firmware Lunii ne sont pas mis à disposition par Lunii SAS ;
4) STUdio étant un logiciel open source, la diffusion des éléments d'interopérabilité (dont la clef de chiffrement) est nécessaire à son fonctionnement.


Lunii SAS dispose du droit exclusif d'effectuer et d'autoriser la traduction, l'adaptation, l'arrangement ou toute autre modification de ses logiciels, **sous réserve des dispositions de l'article L. 122-6-1 du Code de la propriété intellectuelle**.

Article qui définit, en substance, une exception de décompilation (appelée "traduction de la forme du code" dans le texte) dans un but d'interopérabilité. L'exception permet la décompilation **sans autorisation de l'auteur lorsqu'elle est indispensable pour obtenir les informations nécessaires à l'interopérabilité** d'un logiciel créé de façon indépendante.

L'exception s'applique lorsque **trois conditions sont réunies** :
1)  Ces actes sont accomplis **par la personne ayant le droit d'utiliser un exemplaire du logiciel** ou pour son compte par une personne habilitée à cette fin ;
2) Les informations nécessaires à l'interopérabilité **n'ont pas déjà été rendues facilement et rapidement accessibles** aux personnes mentionnées au 1° ci-dessus ;
3) Et ces actes sont **limités aux parties du logiciel d'origine nécessaires à cette interopérabilité**.

**L'utilisation des informations** obtenues par décompilation est également **limitée**. Ces informations ne peuvent être :
1) Ni utilisées **à des fins autres que la réalisation de l'interopérabilité** du logiciel créé de façon indépendante ;
2) Ni **communiquées** à des tiers **sauf si cela est nécessaire à l'interopérabilité du logiciel** créé de façon indépendante ;
3) Ni utilisées pour la mise au point, la production ou la commercialisation d'un logiciel **dont l'expression est substantiellement similaire** ou pour tout autre acte portant atteinte au droit d'auteur.


Je crois fermement que le travail effectué pour STUdio entre dans le cadre de cette exception.


Quelques références sur le sujet :
https://www.legifrance.gouv.fr/codes/article_lc/LEGIARTI000044365559/2023-05-25
https://www.ddg.fr/actualite/decompiler-un-logiciel-enseignements-recents-la-jurisprudence-communautaire

**Reply by fparrel** [reply]
Merci beaucoup pour la réponse. Pour mon cas d'application de lecture sur Android ( https://github.com/fparrel/SDCardStoryTeller ), je pense tomber dans le cadre de l'interopérabilité, donc je saurais quoi répondre si je reçois un recommandé.
